lib/decode/decode.go

```diff
@@ -73,14 +73,19 @@ func translationsForType(to reflect.Type) map[string]string {
 	translations := map[string]string{}
 	for i := 0; i < to.NumField(); i++ {
 		field := to.Field(i)
+		canonKey := strings.ToLower(canonicalFieldKey(field))
+
 		tag, ok := field.Tag.Lookup("alias")
-		if !ok {
-			continue
+		if ok {
+			for _, alias := range strings.Split(tag, ",") {
+				translations[strings.ToLower(alias)] = canonKey
+			}
 		}
 
-		canonKey := strings.ToLower(canonicalFieldKey(field))
-		for _, alias := range strings.Split(tag, ",") {
-			translations[strings.ToLower(alias)] = canonKey
+		// The original key should always be valid too
+		lowerName := strings.ToLower(field.Name)
```

Comment:
Why is the field name always valid? If we've created an alias for it then I would expect only the alias to be valid.

Is this for the case where the source data is coming from a protobuf struct?

Reply:
That comment I left you was for all the changes to key translations.

Originally I was running into issues decoding the JSON serialized from encoding the protobuf struct. The reason there is that I was trying to avoid using gogo protobuf options to customize the JSON struct tag as we will eventually have to stop using gogoproto and I didn't want to introduce more places we have to "fix".

So the protobuf struct ended up like:

`{
   EncryptKey: "...."
}`

However originally in the config, json/hcl/mapstructure struct tags have EncryptKey named just "encrypt" instead so it was failing to decode.

I first started down the path of adding extra aliases to all of these for the actual name of the field. However, after thinking about it, I couldn't see any reason we would have for not accepting the original field name in addition to any alternate names. Especially if we wanted to be able to push down more config via protobufs in the future I was trying to avoid having to add an extra alias to every field we wanted to send that way.

However, all of these changes could be removed now as we translate the protobuf config to the normal config prior to serialization.

I think for now, it might just be best to back out this commit as its not strictly necessary and its not clear that its the best behavior in all the cases.